src/app-layout/drawer/index.tsx

```diff
@@ -159,37 +183,7 @@ export const DrawerTriggersBar = ({
     return 0;
   };
 
-  const overflowItemIsActive = () => {
-    const { overflowItems } = splitItems(drawers?.items, getIndexOfOverflowItem());
-
-    if (drawers && drawers.activeDrawerId) {
-      return overflowItems?.map(item => item.id).indexOf(drawers.activeDrawerId) !== -1;
-    }
-    return false;
-  };
-
-  const getDrawerItems = () => {
-    if (drawers && drawers.items) {
-      const activeIndex = drawers.activeDrawerId && drawers.items.map(item => item.id).indexOf(drawers.activeDrawerId);
-      const lastMainItemIndex = getIndexOfOverflowItem() - 1;
-
-      const activeDrawerItem = drawers.items.find(item => item.id === drawers.activeDrawerId);
-      const lastMainItem = drawers.items.find((item, index) => index === lastMainItemIndex);
-
-      if (activeIndex && overflowItemIsActive() && activeDrawerItem && lastMainItem) {
-        [drawers.items[lastMainItemIndex], drawers.items[activeIndex]] = [
-          drawers.items[activeIndex],
-          drawers.items[lastMainItemIndex],
-        ];
-      }
-
-      return drawers.items;
-    }
-    return [];
-  };
-
-  const { visibleItems, overflowItems } = splitItems(getDrawerItems(), getIndexOfOverflowItem());
-  const overflowItemHasBadge = !!overflowItems?.find(item => item.badge);
+  const { visibleItems, overflowItems } = splitItems(drawers?.items, getIndexOfOverflowItem(), drawers?.activeDrawerId);
```

Comment:
Consolidating splitting by index and active item handing in one function allowed to simplify this code by a lot